packages/mui-material/src/Fab/Fab.js

```diff
@@ -111,28 +104,19 @@ const FabRoot = styled(ButtonBase, {
     }),
   }),
   ({ theme, ownerState }) => ({
-    ...(ownerState.color === 'primary' && {
-      color: theme.palette.primary.contrastText,
-      backgroundColor: theme.palette.primary.main,
-      '&:hover': {
-        backgroundColor: theme.palette.primary.dark,
-        // Reset on touch devices, it doesn't add specificity
-        '@media (hover: none)': {
-          backgroundColor: theme.palette.primary.main,
+    ...(ownerState.color !== 'inherit' &&
+      ownerState.color !== 'default' &&
+      theme.palette[ownerState.color] != null && {
+        color: theme.palette[ownerState.color].contrastText,
```

Comment:
```suggestion
        color: theme.palette[ownerState.color]?.contrastText,
```

I would add optional check everywhere, so that we don't crash if a color that is not supported is added.

Reply:
I have added this check
`theme.palette[ownerState.color] != null`
in the beginning, and it prevents crashing for not supported values. Do you think it is enough?

Reply:
Ah, I have missed it, yep it is enough :)

Reply:
@mnajdova  Thanks for approving the PR. Regarding this comment, when I was working on the Fab component, I also found the same bug that you mentioned here with the Button component. Actually, it will crash whenever it receives an invalid color value. I think it would be good to fix this in the Button component as well. May I create a PR for that?